src/openai/index.ts

```diff
@@ -29,15 +33,26 @@ export default class OpenAIClient {
     this._maxTokens = maxTokens;
   }
 
+  set chatHistoryLength(chatHistoryLength: number) {
+    this._chatHistoryLength = chatHistoryLength;
+  }
+
+  public appendChatMessage(role: string, content: string | undefined): void {
+    this._chatHistory.push({ role: role as string, content: content });
+    if (this._chatHistory.length > this._chatHistoryLength) {
```

Comment:
Shall we call the new setting `_chatHistoryMaxLength` (or maxSize or limit)?

Reply:
Makes more sense probably: i've renamed it to `_chatHistoryMaxLength`